RocketChat/Rocket.Chat.ReactNative
# [FIX] Unwanted clear keychain and request challenges

## Proposed changes

- Fixes the completion of some request challenges. They were return null, and by doing so, the request would not complete. That's the symptom seen on https://github.com/RocketChat/Rocket.Chat.ReactNative/issues/4431
- Removes an unnecessary clear keychain on MMKV patch

## Issue(s)	

- Depends on https://github.com/RocketChat/react-native/pull/5
- Possibly fixes https://github.com/RocketChat/Rocket.Chat.ReactNative/issues/4431

## How to test or reproduce

- As described on #4431, but the infinite loading can be reproduced by running the flow below on Maestro
  - You'll notice the request challenges are fixed (no more infinite loading), but MMKV isn't persisted anymore
  - I didn't want to add a workaround for the clear keychain, since the keychain being cleared is the real issue
- SSL Client Certificate
- Secured Push Notification
- Login with SSO

```yaml
appId: chat.rocket.reactnative
---
- launchApp:
    clearKeychain: true
- tapOn: 'Join our open workspace'
```

## Screenshots

## Types of changes




## Checklist



## Further comments

[NATIVE-236]

[NATIVE-236]: https://rocketchat.atlassian.net/browse/NATIVE-236?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
